### PR TITLE
GitHub Actions release workflow addition

### PR DESCRIPTION
**Pull request by paulmarsicloud**
### Description

- Setting up a `release.yml` GitHub Workflow so that we can publish to NPM in an automated fashion.
- Added `changesets` as the tool to control semantic-release versioning for NPM publishing.
- Added the following in order to add Provenance verification badge on [this NPM page](https://www.npmjs.com/package/react-swipeable)
```
  "publishConfig": {
    "provenance": true
  }
```
- Added a `NPM_TOKEN` to the settings of this repo.
- Amended the `CONTRIBUTING.md` guide

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -106,8 +107,13 @@
     "ts-jest": "^27.1.4",
     "typescript": "^4.6.3"
   },
-  "dependencies": {},
+  "dependencies": {
+    "@changesets/cli": "^2.26.1"
```
Comment (hartzis) [inline package.json:111]: does this need to be a regular dep or can it be a dev-dep?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -80,20 +82,72 @@ $ yarn run format
...
 [warn] Code style issues found in the above file(s). Forgot to run Prettier?
 ```
+
 Then run the formatter:
+
 ```sh
 $ yarn run format
 ```
 
+### Using changesets
```
Comment (hartzis) [inline CONTRIBUTING.md:96]: 🙌 great doc updates!

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -80,20 +82,72 @@ $ yarn run format
...
+3. You don't need to reference the current pull request or whatnot, as that will be added later automatically.
+
+After this, you'll see a new uncommitted file in `.changesets` like:
+
+```sh
+$ git status
+# ....
+Untracked files:
+  (use "git add <file>..." to include in what will be committed)
+	.changeset/flimsy-pandas-marry.md
```
Comment (hartzis) [inline CONTRIBUTING.md:127]: ❓ add a small comment about how/why these are named so randomly?

File: package.json (hunk 1 above)
Comment (paulmarsicloud) [reply package.json:111]: Good call - will move to `devDependencies` 👍

File: CONTRIBUTING.md (hunk 2 above)
Comment (paulmarsicloud) [reply CONTRIBUTING.md:96]: thank you!!

File: CONTRIBUTING.md (hunk 3 above)
Comment (paulmarsicloud) [reply CONTRIBUTING.md:127]: Done!

**Review by hartzis — APPROVED** [review]

**Pull request merged**